torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -3270,7 +3270,8 @@ std::vector<CodeGen::CallArg> TensorExprKernel::prepareRunArgs(
     } else if (input.isDouble()) {
       runArgs.emplace_back(input.toDouble());
     } else if (input.isTensor()) {
-      runArgs.emplace_back(input.toTensor().data_ptr());
+      const auto& t = input.toTensor();
```

Comment:
seems a bit dodgy. If toTensor() returns a Tensor by value, we shouldn't pretend it's a reference. If it returned a const reference then you wouldn't have gotten an error on this line, so I suspect it does return a Tensor by value.